meteor-app/imports/ui/pages/workspace/component.js

```diff
@@ -88,35 +100,31 @@ export default class WorkspacePage extends React.Component {
     toggleLayer(layerIndex, layerVisible);
   }
 
-  _layerOpacityOnChange (event) {
-    const target = event.currentTarget;
-    const layerIndex = parseInt(target.getAttribute('data-layer-index'), 10);
-    const opacity = target.value / 255;
+  _layerOpacityOnChange (context, value) {
+    const opacity = value / 255;
     const {
       updateLayerOpacity,
     } = this.props;
 
-    updateLayerOpacity(layerIndex, opacity);
+    updateLayerOpacity(context['data-layer-index'], opacity);
```

Comment:
So let's talk about the naming issue here first. The first argument passed in is the context passed by the event dispatcher. However, it does not help people read the code by naming it `context`. It's the slider element/component in this case.

OK I thought the slider would be smart to implement a standard HTMLElement interface but if not then sure you don't have `HTMLElement.prototype.getAttribute`. I was trying to say `getAttribute` is the only reliable way of getting attribute values from a `HTMLElement` instance. If you have for example `<a href="abc" something-else="123' />`, the HTML standard only specifies that you should be able to read `element.href`, but the behavior of reading `element['something-else']` is not defined by the standard, even though browsers may (or may not) implement it the same way. The difference here is `href` is a **property** of `HTMLAnchorElement`, while `something-else` is a custom **attribute**.